snippets/csharp/System.Runtime.InteropServices/Marshal/FreeHGlobal/stringtohglobalansi.cs

```diff
@@ -1,34 +1,37 @@
 ﻿//<SNIPPET1>
 using System;
 using System.Runtime.InteropServices;
+using System.Threading;
 
 class MainFunction
 {
     static void Main()
     {
-    Console.WriteLine("\nStringToGlobalAnsi\n");
+        Console.WriteLine("\nStringToGlobalAnsi\n");
 
-    // Create a managed string.
-    String  managedString = "I am a managed String";
-    Console.WriteLine("1) managedString = " + managedString );
+        // Create a managed string.
+        String  managedString = "I am a managed String";
+        Console.WriteLine("1) managedString = " + managedString);
 
-    // Marshal the managed string to unmanaged memory.
-    IntPtr stringPointer = (IntPtr)Marshal.StringToHGlobalAnsi(managedString);
-    Console.WriteLine("2) stringPointer = {0}", stringPointer );
+        // Marshal the managed string to unmanaged memory.
+        IntPtr stringPointer = (IntPtr)Marshal.StringToHGlobalAnsi(managedString);
+        Console.WriteLine("2) stringPointer = {0}", stringPointer);
 
-    // Get the string back from unmanaged memory
-    String RetrievedString = Marshal.PtrToStringAnsi( stringPointer);
-    Console.WriteLine("3) Retrieved from unmanaged memory = " + RetrievedString );
+        // Get the string back from unmanaged memory.
+        String RetrievedString = Marshal.PtrToStringAnsi(stringPointer);
+        Console.WriteLine("3) Retrieved from unmanaged memory = " + RetrievedString);
 
-    // Always free the unmanaged string.
-    Marshal.FreeHGlobal(stringPointer);
+        // Always free the unmanaged string.
+        Marshal.FreeHGlobal(stringPointer);
 
-    // IntPtr handle value is still the same:
-    Console.WriteLine("4) stringPointer = " + stringPointer );
-
-    // However, it contains no data after being freed:
-    String RetrievedString2 = Marshal.PtrToStringAnsi( stringPointer);
-    Console.WriteLine("5) RetrievedString2 = " + RetrievedString2 );
+        // IntPtr handle value is still the same:
+        Console.WriteLine("4) stringPointer = " + stringPointer);
+        
+        // However, the data may be cleared after being freed. Depending on whether the memory allocated to stringPointer
```

Comment:
```suggestion
        // However, the data may be cleared after the memory is freed, depending on whether the memory allocated to stringPointer
```

Reply:
@gewarren  Thanks for improving the comment, I have combined your change into mine as a single commit. Cheers!